addons/l10n_latam_document/models/account_invoice.py

```diff
@@ -0,0 +1,430 @@
+        compute='compute_l10n_latam_next_number',
+        string='Next Number',
+    )
+    l10n_latam_use_documents = fields.Boolean(
+        related='journal_id.l10n_latam_use_documents',
+    )
+    l10n_latam_country_code = fields.Char(
+        related='company_id.country_id.code',
+        help='Technical field used to hide/show fields regarding the '
+        'localization'
+    )
+    display_name = fields.Char(
+        compute='_compute_display_name',
+        string='Document Reference',
+    )
+
+    @api.depends('l10n_latam_document_type_id', 'journal_id')
+    def _compute_l10n_latam_sequence(self):
+        for rec in self:
+            rec.l10n_latam_sequence_id = \
+                rec.journal_id.get_document_type_sequence(rec)
+
+    @api.depends(
+        'amount_untaxed', 'amount_tax', 'tax_line_ids', 'l10n_latam_document_type_id')
+    def _compute_l10n_latam_amount_and_taxes(self):
+        for invoice in self:
+            taxes_included = (
+                invoice.l10n_latam_document_type_id and
+                invoice.l10n_latam_document_type_id.get_taxes_included() or False)
+            if not taxes_included:
+                l10n_latam_amount_tax = invoice.amount_tax
+                l10n_latam_amount_untaxed = invoice.amount_untaxed
+                not_included_taxes = invoice.tax_line_ids
+            else:
+                included_taxes = invoice.tax_line_ids.filtered(
+                    lambda x: x.tax_id in taxes_included)
+                not_included_taxes = (
+                    invoice.tax_line_ids - included_taxes)
+                l10n_latam_amount_tax = sum(not_included_taxes.mapped('amount'))
+                l10n_latam_amount_untaxed = invoice.amount_untaxed + sum(
+                    included_taxes.mapped('amount'))
+            invoice.l10n_latam_amount_tax = l10n_latam_amount_tax
+            invoice.l10n_latam_amount_untaxed = l10n_latam_amount_untaxed
+            invoice.l10n_latam_tax_line_ids = not_included_taxes
+
+    @api.depends(
+        'journal_id.sequence_id.number_next_actual',
+        'l10n_latam_sequence_id.number_next_actual',
+    )
+    def compute_l10n_latam_next_number(self):
+        """ Show next number only for invoices without number and on draft
+        state """
+        for invoice in self.filtered(
+                lambda x: not x.display_name and x.state == 'draft'):
+            if invoice.l10n_latam_use_documents:
+                sequence = invoice.l10n_latam_sequence_id
+            elif (
+                    invoice.type in ['out_refund', 'in_refund'] and
+                    invoice.journal_id.refund_sequence
+            ):
+                sequence = invoice.journal_id.refund_sequence_id
+            else:
+                sequence = invoice.journal_id.sequence_id
+            # we must check if sequence use date ranges
+            if not sequence.use_date_range:
+                invoice.l10n_latam_next_number = sequence.number_next_actual
```

Comment:
```suggestion
                invoice.l10n_latam_next_number = sequence.number_next_actual
                continue
```
To avoid the next else sentence